docs/src/main/tut/ssl-tls.md

```diff
@@ -0,0 +1,167 @@
+    }
+
+}
+
+object handlers {
+
+    class ServerRPCService[F[_]: Effect] extends RPCService[F] {
+    	def unary(a: A): F[C] = Effect[F].delay(c1)
+    }
+
+}
+
+trait Runtime {
+
+	import service._
+    import handlers._
+
+    implicit val freesRPCHandler: ServerRPCService[ConcurrentMonad] =
+      new ServerRPCService[ConcurrentMonad]
+
+    // First of all, we have to load the certs into files. These files has to be locally in our
+    // module in the resources folder.
+
+    val serverCertFile: File                         = TestUtils.loadCert("server1.pem")
+    val serverPrivateKeyFile: File                   = TestUtils.loadCert("server1.key")
+    val serverTrustedCaCerts: Array[X509Certificate] = Array(TestUtils.loadX509Cert("ca.pem"))
+
+    // We have to build the SslContext passing our server certificates, configuring the OpenSSL
+    // and requiring the client auth.
+
+    val serverSslContext: SslContext =
+      GrpcSslContexts
+        .configure(
+          GrpcSslContexts.forServer(serverCertFile, serverPrivateKeyFile),
+          SslProvider.OPENSSL)
+        .trustManager(serverTrustedCaCerts: _*)
+        .clientAuth(ClientAuth.REQUIRE)
+        .build()
+
+    // Adding to the GrpConfig our SslContext with SetSslContext, when we will create implicitly our 
+    // server with ServerW.netty, our server will require SSL encryption in the client side.
+
+    val grpcConfigs: List[GrpcConfig] = List(
+      SetSslContext(serverSslContext),
+      AddService(RPCService.bindService[ConcurrentMonad])
+    )
+
+    // Important. We have to create the server with Netty. OkHttp is not supported for the Ssl 
+    // encryption in frees-rpc.
+
+    implicit val serverW: ServerW = ServerW.netty(SC.port, grpcConfigs)
+
+}
+
+object implicits extends Runtime
+
+```
+
+
+Lastly, as we did before with the server side, let's see what happens on the client side.
+
+```tut:silent
+
+import freestyle.rpc.client.OverrideAuthority
+import freestyle.rpc.client.netty.{
+  NettyChannelInterpreter,
+  NettyNegotiationType,
+  NettySslContext,
+  NettyUsePlaintext
+}
+
+object client {
+
+    // First of all, we have to load the certs into files.
+
+    val clientCertChainFile: File                    = TestUtils.loadCert("client.pem")
+    val clientPrivateKeyFile: File                   = TestUtils.loadCert("client.key")
+    val clientTrustedCaCerts: Array[X509Certificate] = Array(TestUtils.loadX509Cert("ca.pem"))
+
+    // We have to create the SslContext for the client like as we did in the server.
+
+    val clientSslContext: SslContext =
+      GrpcSslContexts.forClient
+        .keyManager(clientCertChainFile, clientPrivateKeyFile)
+        .trustManager(clientTrustedCaCerts: _*)
+        .build()
+}
+
+object MainApp {
+
+	import client._
+
+	// Important, the channel interpreter have to be NettyChannelInterpreter.
+
+	// In this case, we are creating the channel interpreter with a specifics ManagedChannelConfig
```

Comment:
// In this case, we are creating the channel interpreter with a specific ManagedChannelConfig